File: src/services/default_model.ts
import type { DefaultModelSetting } from '@/types/default-model';
import { invoke } from '@tauri-apps/api/core';

export async function listDefaultModelSettings(): Promise<DefaultModelSetting[]> {
  return invoke<DefaultModelSetting[]>('list_default_model_settings');
}

export async function upsertDefaultModelSetting(setting: DefaultModelSetting): Promise<string> {
  return invoke<string>('upsert_default_model_setting', { setting });
}

File: src-tauri/src/commands/default_model.rs
use crate::db::local_store::LocalJsonStore;
use crate::db::repositories::default_model::DefaultModelSettingRepository;
use crate::models::default_model::DefaultModelSetting;

#[tauri::command]
pub fn list_default_model_settings(
    store: tauri::State<'_, LocalJsonStore>,
) -> Result<Vec<DefaultModelSetting>, String> {
    let repo = DefaultModelSettingRepository::new(&store);
    repo.list()
}

#[tauri::command]
pub fn upsert_default_model_setting(
    store: tauri::State<'_, LocalJsonStore>,
    setting: DefaultModelSetting,
) -> Result<String, String> {
    let repo = DefaultModelSettingRepository::new(&store);
    repo.upsert(&setting)
}
